.gitlab-ci.yml

```diff
@@ -1,16 +1,19 @@
 image: node:12.18.3
 
 stages:
-  - build
   - test
 
-build:
-  stage: build
-  script:
-    - node -v
-    - npm install
+cache:
+  paths:
+    - node_modules/
 
-test:
+tests:
```

Comment:
Line # 10 still has tests, this is appearing inconsistent as your defined stage is test. Nevertheless it can be cleaned up in the next PR easily if you deem it fit. For the PR looks solid and is approved.

Reply:
The convention have the stage be 'test'. The job name can be anything and I felt 'tests' fit as I would consider my models more of a 'unit test' and the tests for my API an integration test. I could split my tests into multiple jobs called 'test_a' and 'test_b' but I would use the stage 'test' for both. This would make the Testing report a lot easier to read as you would see a clear separation in the gitlab testing tab UI however, since I opted to a faster runtime to run both my testing scripts in the same runner, I decided to name it tests.

In the future, I plan on separating it into different jobs.... 'unit_tests', 'integration_tests', & 'e2e_tests'.

Reply:
Alternatively, I could also separate them into what my files intended them to be... 'db_tests' & 'api_tests'... etc.

As for the job naming convention, I'm not sure if there is one.